tensorflow/lite/micro/memory_helpers.cc

```diff
@@ -90,6 +90,9 @@ TfLiteStatus TfLiteTypeSizeOf(TfLiteType type, size_t* size) {
     case kTfLiteComplex128:
       *size = sizeof(double) * 2;
       break;
+    case kTfLiteInt4:
```

Comment:
let's add a comment describing why describing `kTfLiteInt4` has size of 1 byte is what we want.